Source/WebCore/PAL/pal/spi/cocoa/WritingToolsSPI.h

```diff
@@ -34,25 +37,170 @@ DECLARE_SYSTEM_HEADER
 #import <WritingTools/WTSession_Private.h>
 #import <WritingTools/WritingTools.h>
 
-#if PLATFORM(MAC)
+#else
 
-using PlatformWritingToolsBehavior = NSWritingToolsBehavior;
+#import <Foundation/Foundation.h>
 
-constexpr auto PlatformWritingToolsBehaviorNone = NSWritingToolsBehaviorNone;
-constexpr auto PlatformWritingToolsBehaviorDefault = NSWritingToolsBehaviorDefault;
-constexpr auto PlatformWritingToolsBehaviorLimited = NSWritingToolsBehaviorLimited;
-constexpr auto PlatformWritingToolsBehaviorComplete = NSWritingToolsBehaviorComplete;
+extern NSAttributedStringKey const WTWritingToolsPreservedAttributeName;
 
-using PlatformWritingToolsResultOptions = NSWritingToolsResultOptions;
+typedef NS_ENUM(NSInteger, WTRequestedTool) {
+    WTRequestedToolIndex = 0,
 
-constexpr auto PlatformWritingToolsResultPlainText = NSWritingToolsResultPlainText;
-constexpr auto PlatformWritingToolsResultRichText = NSWritingToolsResultRichText;
-constexpr auto PlatformWritingToolsResultList = NSWritingToolsResultList;
-constexpr auto PlatformWritingToolsResultTable = NSWritingToolsResultTable;
+    WTRequestedToolProofread = 1,
+    WTRequestedToolRewrite = 2,
+    WTRequestedToolRewriteProofread = 3,
+
+    WTRequestedToolRewriteFriendly = 11,
+    WTRequestedToolRewriteProfessional = 12,
+    WTRequestedToolRewriteConcise = 13,
+    WTRequestedToolRewriteOpenEnded = 19,
+
+    WTRequestedToolTransformSummary = 21,
+    WTRequestedToolTransformKeyPoints = 22,
+    WTRequestedToolTransformList = 23,
+    WTRequestedToolTransformTable = 24,
+
+    WTRequestedToolSmartReply = 101,
+
+    WTRequestedToolCompose = 201,
+};
+
+// MARK: WTContext
+
+@protocol WTTextViewDelegate;
+
+@interface WTContext : NSObject
+
+@property (nonatomic, readonly) NSUUID *uuid;
+
+@property (nonatomic, readonly) NSAttributedString *attributedText;
+
+@property (nonatomic) NSRange range;
+
+- (instancetype)initWithAttributedText:(NSAttributedString *)attributedText range:(NSRange)range;
+
+@end
+
+// MARK: WTSession
+
+typedef NS_ENUM(NSInteger, WTCompositionSessionType) {
+    WTCompositionSessionTypeNone,
+    WTCompositionSessionTypeMagic,
+    WTCompositionSessionTypeFriendly,
+    WTCompositionSessionTypeProfessional,
+    WTCompositionSessionTypeConcise,
+    WTCompositionSessionTypeOpenEnded,
+    WTCompositionSessionTypeSummary,
+    WTCompositionSessionTypeKeyPoints,
+    WTCompositionSessionTypeList,
+    WTCompositionSessionTypeTable,
+    WTCompositionSessionTypeCompose,
+    WTCompositionSessionTypeSmartReply,
+    WTCompositionSessionTypeProofread,
+};
```

Comment:
Same with these cases, many are not referred to directly in our project

Reply:
for this, I realize we're not using many, but I was hesitant about changing them since that would affect their ordinal number, so I didn't want to risk accidentally changing one of their numeric values